File: javascript/hamming/hamming.js
function Hamming() {}

Hamming.prototype.compute = function (one, two) {
  if (one.length !== two.length) {
    throw new Error('DNA strands must be of equal length.');
  }

  const split1 = one.split('');
  const split2 = two.split('');
  let diffHamm = 0;

  for (let i = 0; i < one.length; i++) {
    if (split1[i] !== split2[i]) {
      diffHamm++;
    }
  }

  return diffHamm;
};

module.exports = Hamming;

File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

namespace NucleoTidesOfMan
{
    public static class Hamming
    {
        public static int Compute(string strand1, string strand2)
        {
            return strand1.Zip(strand2, (one, two) =>
                one.Equals(two) ? 0 : 1).Sum();
        }
    }
}
